src/docs/asciidoc/advancedFeatures.adoc

```diff
@@ -716,9 +734,112 @@ as the library is usually upstream of the client code.
+is used instead.
+<3> The resulting `Mono<String>` thus emits `"Hello Stranger"`.
+
+Similarly, in case of several attempts to write the same key to the `Context`, the
+**relative order of the writes** matters too: operators reading the `Context` will see
+the value that was set closest to under them, as demonstrated in the following example:
+
+[source,java]
+----
+String key = "key";
+Mono<String> r = Mono.just("Hello")
+                .flatMap( s -> Mono.subscriberContext()
+                                   .map( ctx -> s + " " + ctx.get(key)))
+                .subscriberContext(ctx -> ctx.put(key, "Reactor")) //<1>
+                .subscriberContext(ctx -> ctx.put(key, "World")); //<2>
+
+StepVerifier.create(r)
+            .expectNext("Hello Reactor") //<3>
+            .verifyComplete();
+----
+<1> A write attempt on key `"key"`.
+<2> Another write attempt on key `"key"`.
+<3> The `map` only saw the value set closest to it (and below it): `"Reactor"`.
+
+Here what happens is that the `Context` is populated during subscription with `"World"`.
+Then the subscription signal moves upstream, and another write happens. This produces a
+second immutable `Context` with a value of `"Reactor"`. After that, data starts flowing.
+The `flatMap` sees the `Context` closest to it, which is our second `Context` with the
+`"Reactor"` value.
+
+You might wonder if the `Context` is propagated along with the data signal. If that was
+the case, putting another `flatMap` between these two writes would use the value from
+the top `Context`. But this is not the case, as demonstrated by the following example:
+
+[source,java]
+----
+String key = "key";
+Mono<String> r = Mono.just("Hello")
+                     .flatMap( s -> Mono.subscriberContext()
+                                        .map( ctx -> s + " " + ctx.get(key))) //<3>
+                     .subscriberContext(ctx -> ctx.put(key, "Reactor")) //<2>
+                     .flatMap( s -> Mono.subscriberContext()
+                                        .map( ctx -> s + " " + ctx.get(key))) //<4>
+                     .subscriberContext(ctx -> ctx.put(key, "World")); //<1>
+
+StepVerifier.create(r)
+            .expectNext("Hello Reactor World") //<5>
+            .verifyComplete();
+----
+<1> This is the first write to happen.
+<2> This is the second write to happen.
+<3> First `flatMap` sees second write.
+<4> Second `flatMap` concatenates result from first one with the value from **first write**.
+<5> The `Mono` emits `"Hello Reactor World"`.
+
+The reason is that the `Context` is associated to the `Subscriber` and each operator
+accesses the `Context` by requesting it from its downstream `Subscriber`.
+
+=== Full Example
+Let's consider a more real life example of a library reading information from the `Context`:
+A reactive HTTP client that takes a `Mono<String>` as the source of data for a `PUT` but
+also looks for a particular Context key to add a correlation ID to the request's headers.
 
```

Comment:
It might be good to add a sample that discusses this:

```java
String key = "message";
Mono<?> r = Mono.just("Hello")
	.flatMap( s -> Mono.subscriberContext()
		.map( ctx -> s + " " + ctx.get(key))
	)
	.flatMap( s -> Mono.subscriberContext()
		.map( ctx -> s + " " + ctx.get(key))
		// subscriberContext does not impact anything outside of flatMap
		.subscriberContext(ctx -> ctx.put(key, "Reactor"))
	)
	.subscriberContext(ctx -> ctx.put(key, "World"));

StepVerifier.create(r)
	.expectNext("Hello World Reactor")
	.verifyComplete();
```

Reply:
One more perhaps meaningful example

```java

String key = "message";

Mono<?> r = Mono.subscriberContext()
	// below does not impact Mono.subscriberContext because it is immutable
	.map( ctx -> ctx.put(key, "Hello"))
	.flatMap( ctx -> Mono.subscriberContext())
	.map( ctx -> ctx.getOrDefault(key,"Default"));

StepVerifier.create(r)
	.expectNext("Default")
	.verifyComplete();
```